src/main/java/ether/EtherClient.java

```diff
@@ -0,0 +1,719 @@
+        //System.out.println("The request string in getEtherBalance is "+requestString);
+        String etherBalance;
+        try {
+            etherBalance=(String)httpAgent.getHttpResponse(queryEtherBalanceString);
+        }catch (IOException e) {
+            if(Settings.DEBUG_INFO){
+                System.out.println("Failed to query the Ether Balance for: " + address);
+            }
+            throw e;
+        }
+
+        return etherBalance;
+    }
+
+
+    /**
+     * Get the balance of Tokens on the Ethereum network.
+     * @param address the address in the Ethereum Network
+     * @param  httpAgent The Http wrapper
+     * @return the Tokens balance
+     * @throws IOException
+     */
+    public static String getTokenBalance(String address, Http httpAgent) throws IOException{
+        CallTransaction.Function balanceOf = Settings.TOKEN_CONTRACT.getByName("balanceOf");
+        byte [] functionBytes=balanceOf.encode(address);
+        String requestString = "{\"method\":\"eth_call\"," +
+                "\"params\":[" +
+                "{" +
+                "\"to\":\""+Settings.TOKEN_ADDRESS+"\"," +
+                "\"data\":\""+"0x" + new String(org.apache.commons.codec.binary.Hex.encodeHex(functionBytes))+"\"" +
+                "}," +
+                "\"latest\"" +
+                "]," +
+                "\"id\":42,\"jsonrpc\":\"2.0\"}";
+        if(Settings.DEBUG_INFO) {
+            System.out.println("Request in getTokenBalance = "+requestString);
+        }
+        String tokenBalance;
+        try {
+            tokenBalance=(String)httpAgent.getHttpResponse(requestString);
+        }catch (IOException e) {
+            throw e;
+        }
+
+        if(Settings.DEBUG_INFO) {
+            System.out.println("Balance of " + address + " = " + new Float(new BigInteger(tokenBalance.substring(2), 16).doubleValue() / (new BigInteger(Settings.APPENDING_ZEROS_FOR_TOKEN, 10).doubleValue())).toString() + " TKN");
+        }
+
+        return tokenBalance;
+    }
+
+
+    /**
+     * Gets the payment channel info by Sender and Receiver addresses.
+     * @param senderAddress The sender address in the Ethereum Network
+     * @param receiverAddress The receiver address in the Ethereum Network
+     * @param httpAgent Http wrapper
+     * @return PaymentChannel object that holds all channel info.
+     * @throws DecoderException
+     * @throws IOException
+     */
+    public static EtherUtility.PaymentChannel getChannelInfo(String senderAddress, String receiverAddress, Http httpAgent) throws DecoderException, IOException {
+
+        byte[] keyInBytes= EtherUtility.getChannelHash(senderAddress, receiverAddress);
+
+        CallTransaction.Function channels = Settings.CHANNEL_CONTRACT.getByName("channels");
+        byte [] functionBytes=channels.encode(keyInBytes);
+        String requestString = "{\"method\":\"eth_call\"," +
```

Comment:
If you use this request string in multiple places, it might help to make one functino with the string in it and pass in the addresses and data as parameters.